SU-HSDO/suhumsci
# SHS-5905: Rework social media footer

# READY FOR REVIEW

## Summary
- Adds a social media block as a custom block plugin
- Adds permission for site managers to edit the social media block
- Adds an exception to allow the contextual link to work for site managers

## Need Review By (Date)
2024-10-23

## Urgency
low

## Steps to Test
1. Login to the site as an admin
2. Visit the block layout page `/admin/structure/block`
3. Under the footer region, click "Place block"
4. Click "Place Block next to the "Social Media Block"
5. Uncheck "Display title"
6. Set some options for the Icon Size and the Layout
7. Add some links into the form - test only adding a link and not a label (allowed), and test adding a label and not a link (not allowed)
8. Save the block
9. View on the front-end
10. Verify that the social media block shows up
11. Masquerade as a user with only site manager role
12. View the front-end
13. Verify that a site manager can edit, but not remove, the new social block

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208772053918607
